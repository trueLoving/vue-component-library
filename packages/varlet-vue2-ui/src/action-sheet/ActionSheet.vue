<template>
  <var-popup
    class="var-action-sheet__popup-radius"
    position="bottom"
    :overlay="overlay"
    :overlay-class="overlayClass"
    :overlay-style="overlayStyle"
    :lock-scroll="lockScroll"
    :close-on-click-overlay="closeOnClickOverlay"
    :teleport="teleport"
    :show="popupShow"
    @open="$emit('open')"
    @close="$emit('close')"
    @closed="$emit('closed')"
    @opened="$emit('opened')"
    @route-change="$emit('route-change')"
    @update:show="(value) => $emit('update:show', value)"
  >
    <div class="var-action-sheet var--box" v-bind="$attrs">
      <slot name="title">
        <div class="var-action-sheet__title">{{ dt(title, pack.actionSheetTitle) }}</div>
      </slot>

      <slot name="actions">
        <div
          class="var-action-sheet__action-item"
          v-for="action in actions"
          :class="[action.className, action.disabled ? 'var-action-sheet--disabled' : null]"
          v-ripple="{ disabled: action.disabled }"
          :key="action.name"
          :style="{ color: action.color }"
          @click="handleSelect(action)"
        >
          <var-icon
            class="var-action-sheet__action-icon"
            var-action-sheet-cover
            :name="action.icon"
            :size="action.iconSize"
            v-if="action.icon"
          />
          <div class="var-action-sheet__action-name">{{ action.name }}</div>
        </div>
      </slot>
    </div>
  </var-popup>
</template>

<script>
import Ripple from '../ripple'
import VarPopup from '../popup'
import VarIcon from '../icon'
import { defineComponent } from '../utils/create'
import { props } from './props'
import { dt } from '../utils/shared'
import { pack } from '../locale'

export default defineComponent({
  name: 'VarActionSheet',

  directives: { Ripple },

  components: {
    VarPopup,
    VarIcon,
  },

  inheritAttrs: false,

  props,

  data: () => ({
    popupShow: false,
  }),

  computed: {
    pack() {
      return pack.value
    },
  },

  watch: {
    show: {
      handler(newValue) {
        this.popupShow = newValue
      },
      immediate: true,
    },
  },

  methods: {
    dt,

    handleSelect(action) {
      if (action.disabled) {
        return
      }

      this.$emit('select', action)

      this.closeOnClickAction && this.$emit('update:show', false)
    },
  },
})
</script>

<style lang="less">
@import '../styles/common';
@import '../icon/icon';
@import '../ripple/ripple';
@import '../popup/popup';
@import './actionSheet';
</style>
