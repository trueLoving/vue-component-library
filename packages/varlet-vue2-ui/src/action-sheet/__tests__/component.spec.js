import ActionSheet from '../index'
import VarActionSheet from '../ActionSheet'
import { mount } from '@vue/test-utils'
import Vue from 'vue'

test('test action-sheet component plugin', () => {
  Vue.use(ActionSheet.Component)
  expect(Vue.component(ActionSheet.Component.name)).toBeTruthy()
})

test('test action-sheet styles', () => {
  const Wrapper = {
    components: {
      [VarActionSheet.name]: VarActionSheet,
    },
    data: () => ({
      actions: [
        {
          name: 'Item 01',
          icon: 'account-circle',
          iconSize: '30px',
          color: '#4caf50',
          className: 'customClass',
        },
        {
          name: 'Item 02',
          icon: 'account-circle',
          disabled: true,
        },
      ],
    }),
    template: `
      <var-action-sheet :actions="actions" />
    `,
  }
  const wrapper = mount(Wrapper)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})
