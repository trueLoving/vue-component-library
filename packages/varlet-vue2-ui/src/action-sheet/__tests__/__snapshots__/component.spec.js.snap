// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test action-sheet styles 1`] = `
"<transition-stub name=\\"var-fade\\" class=\\"var-action-sheet__popup-radius\\">
  <div class=\\"var--box var-popup\\" style=\\"z-index: 1998; display: none;\\">
    <div class=\\"var-popup__overlay\\" style=\\"z-index: 1999;\\"></div>
    <transition-stub name=\\"var-pop-bottom\\"></transition-stub>
  </div>
</transition-stub>"
`;
